### DISCUSSION TITLE
How to cancel mutation in v3?

### DISCUSSION THREAD

**Discussion by qkreltms** [TanStack/query#1560] [Q&A] [answered]
key is unnecessary when using useMutation, but by canceling queryClient.cancelQueries it must be provided.

I want to cancel just one mutation, but seems queryClient.cancelQueries() cancels all requests if key is not provided.

**Comment by TkDodo** [comment, marked as answer]
This was just asked 3 days ago: https://github.com/tannerlinsley/react-query/discussions/1551
